### ISSUE TITLE
Slide unsubscribing from subreddits?

### ISSUE THREAD

**Issue by rmenessec** [ccrama/Slide#1436]
Slide version: 5.1.3
Android version: Nexus 10, Google stock 5.1.1 image (LMY49H); Nexus 6 Google stock 6.0.1 image (MMB29X)

I have a relatively large number of subscribed subreddits; maybe 30+. Slide "imported" all of my subscribed subreddits to the tab bar and sidebar when I first installed it. Since then, three subreddits have vanished. After some poking around, it turns out that I've become unsubscribed - or Slide thinks I have been.

I didn't think to verify with a browser or another Android client before re-subscribing the "missing" subreddits, so I can't verify what happened. I suppose it's more likely that Slide got confused about my subscriptions, as opposed to Slide unsubscribing me. Unfortunately, I don't know for certain which it was.

**Comment by ccrama** [comment]
The only way for that to occurs is if you removed it in Settings > Reorder or if you unchecked "subscribe" in the subreddit sidebar. They are stored in a comma separated list, not really a way  Slide could get "confused" about it without a user telling it to do something in one way or another

**Comment by rmenessec** [comment]
Definitely did not accidentally uncheck three subscription boxes in three separate subreddits after accidentally opening the sidebar. ;)

The only thing I've done in Reorder is... reorder. I tapped the alpha-sort button a couple of times to see if I could switch between ascending / descending sort. :D I haven't done any drag-to-sort. The only time I've done a multi-select is just now, to see exactly what options might be in Reorder.

One other possibility: I've experienced a few crashes with Slide 5.x while having connectivity problems. Sometimes, when content is unreachable, Slide crashes instead of producing an error or just timing out canvas elements. It's possible I was in Reorder when one of these happened.

**Comment by ccrama** [comment]
It still wouldn't have saved, and if the subs weren't on the extraneous ends (front or back), there is no way the CSV would have removed an item in the middle. Are you sure you didn't unsubscribe from it anywhere else and possibly hit the sync button? Or were the subs at the start or end of the list?

**Comment by rmenessec** [comment]
I was about to open an enhancement request, but I _did_ just mess up with the image captions "bug" - I think I'd better ask first. Is there a way to get something like Now for Reddit's default multi-column view? It would be nice to have one column displaying a list of posts and the other column showing a full post after selecting it from the list.

**Comment by ccrama** [comment]
No, that's not possible on Slide and probably won't do that for the future (multicolumn already has the popup mode, and this would also interfere with the comment pane layout mode). Sorry for the inconvenience!

**Comment by rmenessec** [comment]
Hm. No, the subreddits that vanished were in the middle of the list. I didn't even know it was possible to remove subreddits from the Reorder task until just now. I would definitely remember deleting three subreddits from the sidebar views - it's a lot of work.

**Comment by ccrama** [comment]
Well let me know if you can reproduce this. I haven't gotten any other reports of this and don't want to say it's user error, but if I can't reproduce it I can't fix it :(

Thanks for the report!

**Comment by rmenessec** [comment]
That's reasonable. Feel free to close, and I'll reopen if I run into this again.

Does Slide have a "send logs" feature that I missed? A debug mode? If there's a way to capture logs, I'd like to start it up now, so that I can send something concrete if it happens again.

**Comment by ccrama** [comment]
Sounds good! 

And when the app crashes, it sends a log if you click the "report" button. Otherwise, the only other way to get logs is ADB on your computer. There's great integration with the logs in Android Studio, and if you build the app yourself it creates a second "debug" version of the app on your phone (if interested). If you need help setting it up let me know!

**Comment by rmenessec** [comment]
You're referring to logcat, right? I can grab that. Know of any logcat "sanitizer" apps? I've used a few logcat viewers, and they're lucky if they can so much as color-code by line, much less look for and remove device IDs or phone numbers—that sort of thing. 😁

Not sure I want to mess with Android Studio right now. Last I looked at the SDK alone, it was huge, gobbled RAM, and still didn't have native 64-bit binaries; even for Linux hosts. Has any of that changed? The requirements page still mentioned i386 libs the last I looked at it...

The offer is appreciated, either way. 😉
